### PR TITLE
Please add arm template

### PR DESCRIPTION
**Pull request by denniszielke**
ARM templates are easier for IT departments to deploy.

### CODE REVIEW

File: arm/openshift_template.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          "networkProfile": {
+            "vnetCidr": "10.0.0.0/8"
+          },
+          "authProfile": {
+            "identityProviders": [
+              {
+                "name": "Azure AD",
+                "provider": {
+                  "kind": "AADIdentityProvider",
+                  "clientId": "[parameters('servicePrincipalClientId')]",
```
Comment (sozercan) [inline arm/openshift_template.json]: this is AAD client ID and password, doesn't need to be a service principal

File: arm/openshift_template.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            "count": 3,
+            "subnetCidr": "10.0.0.0/24",
+            "vmSize": "Standard_D4s_v3",
+            "osType": "Linux"
+          },
+          "agentPoolProfiles": [
+            {
+              "role": "compute",
+              "name": "compute",
+              "count": "[parameters('agentCount')]",
```
Comment (sozercan) [inline arm/openshift_template.json]: `computeCount`

File: docs/openshift-on-azure-walkthrough.md [hunk 3]
```diff
@@ -71,6 +71,32 @@ az group create --name $CLUSTER_NAME --location $LOCATION
...
+
+az group deployment create \
+    --name openshiftmanaged \
+    --resource-group $CLUSTER_NAME \
+    --template-file "arm/openshift_template.json" \
+    --parameters "arm/openshift_parameters.json" \
+    --parameters "resourceName=$CLUSTER_NAME" \
+        "location=$LOCATION" \
+        "fqdn=$FQDN" \
+        "servicePrincipalClientId=$AAD_ID" \
```
Comment (sozercan) [inline docs/openshift-on-azure-walkthrough.md]: same here for servicePrincipalClientId and servicePrincipalClientSecret

File: arm/openshift_template.json (hunk 2 above)
Comment (TylerMSFT) [reply arm/openshift_template.json]: What's an 'agentCount'?

Comment (TylerMSFT) [reply arm/openshift_template.json]: Do you want to call it that, then?  Maybe its a domain knowledge thing that I'm missing, but 'agentCount' doesn't mean anything to me.

**Comment by sozercan** [comment]
@sabbaour

File: arm/openshift_template.json [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            "type": "securestring"
+        }
+    },
+    "resources": [
+      {
+        "location": "[parameters('location')]",
+        "name": "[parameters('resourceName')]",
+        "type": "Microsoft.ContainerService/openShiftManagedClusters",
+        "apiVersion": "2018-09-30-preview",
```
Comment (julienstroheker) [inline arm/openshift_template.json]: This should be changed to `2019-04-30`

File: arm/openshift_template.json [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                "description": "The location of cluster resource."
+            }
+        },
+        "fqdn": {
+            "type": "string",
+            "metadata": {
+                "description": "Fully qualified dns of the openshift cluster"
+            }
+        },
+        "agentCount": {
```
Comment (julienstroheker) [inline arm/openshift_template.json]: computeCount

**Pull request closed without merging**